sdk/eventhub/event-hubs/src/managementClient.ts

```diff
@@ -306,31 +310,102 @@ export class ManagementClient extends LinkEntity {
-        maxRetries: options.retryOptions && options.retryOptions.maxRetries,
-        abortSignal: options.abortSignal,
-        requestName: options.requestName,
-        timeoutInSeconds: getRetryAttemptTimeoutInMs(options.retryOptions) / 1000,
-        delayInSeconds:
-          options.retryOptions &&
-          options.retryOptions.retryInterval &&
-          options.retryOptions.retryInterval >= 0
-            ? options.retryOptions.retryInterval / 1000
-            : undefined
+      const sendOperationPromise = () =>
+        new Promise<Message>(async (resolve, reject) => {
+          try {
+            let count = 0;
+
+            if (!options) {
+              options = {};
+            }
+
+            const retryTimeoutInMs = getRetryAttemptTimeoutInMs(options.retryOptions);
+            let timeTakenByInit = 0;
+
+            if (!this._isMgmtRequestResponseLinkOpen()) {
+              log.mgmt(
+                "[%s] Acquiring lock to get the management req res link.",
+                this._context.connectionId
+              );
+
+              const initOperationStartTime = Date.now();
+
+              const actionAfterTimeout = () => {
+                const desc: string = `The request with message_id "${request.message_id}" timed out. Please try again later.`;
+                const e: Error = {
+                  name: "OperationTimeoutError",
+                  message: desc
+                };
+
+                return reject(translate(e));
+              };
+
+              const waitTimer = setTimeout(actionAfterTimeout, retryTimeoutInMs);
+
+              try {
+                await defaultLock.acquire(this.managementLock, () => {
+                  return this._init();
+                });
+              } catch (err) {
+                return reject(translate(err));
+              } finally {
+                clearTimeout(waitTimer);
+              }
+              timeTakenByInit = Date.now() - initOperationStartTime;
+            }
+
+            const remainingOperationTimeoutInMs = retryTimeoutInMs - timeTakenByInit;
+
+            const sendRequestOptions: SendRequestOptions = {
+              abortSignal: options.abortSignal,
+              requestName: options.requestName,
+              timeoutInSeconds: remainingOperationTimeoutInMs / 1000
+            };
+
+            count++;
+            if (count !== 1) {
+              // Generate a new message_id every time after the first attempt
+              request.message_id = generate_uuid();
+            } else if (!request.message_id) {
+              // Set the message_id in the first attempt only if it is not set
+              request.message_id = generate_uuid();
+            }
+
+            const result = await this._mgmtReqResLink!.sendRequest(request, sendRequestOptions);
+            resolve(result);
+          } catch (err) {
+            err = translate(err);
+            const address = this._mgmtReqResLink!.sender.address || "address";
```

Comment:
To be extra safe, I think we should check that `this._mgmtReqResLink` actually exists. For instance, if a customer called `close()` right after making a management request, `this._mgmtReqResLink` would not exist.